static/js/settings_org.js

```diff
@@ -684,6 +684,14 @@ exports.build_page = function () {
         // they do, we transition to the "unsaved" state showing the
         // save/discard widget; otherwise, we hide that widget (the
         // "discarded" state).
+
+        if ($(e.target).hasClass("no-input-change-detection")) {
+            // This is to prevent input changes detection in elements
+            // within a subsection whose changes should not affect the
+            // visibility of the discard button
+            return false;
+        }
```

Comment:
fixed the issue mentioned by @timabbott by fixing two subtle bugs after putting many print statements and found that few unnecessary event handlers were working there.